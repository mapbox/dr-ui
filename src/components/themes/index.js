import main from './themes';

export default main;
