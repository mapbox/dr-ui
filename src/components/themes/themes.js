import React from 'react';
import PropTypes from 'prop-types';
import Icon from '@mapbox/mr-ui/icon';

class Image extends React.Component {
  render() {
    const { color, icon } = this.props;
    return (
      <div
        className={`bg-${color}-light round-full color-${color} flex-parent flex-parent--center-main flex-parent--center-cross`}
        style={{ width: 50, height: 50 }}
      >
        <Icon name={icon} size={40} />
      </div>
    );
  }
}

Image.propTypes = {
  color: PropTypes.oneOf([
    'gray',
    'orange',
    'red',
    'green',
    'purple',
    'pink',
    'yellow',
    'blue'
  ]).isRequired,
  icon: PropTypes.string
};

const themes = {
  default: {
    image: <Image icon="book" color="gray" />,
    label: 'Note',
    icon: 'book',
    styles: {
      background: '#f4f7fb',
      color: '#425870'
    }
  },
  warning: {
    image: <Image icon="alert" color="orange" />,
    label: 'Warning',
    icon: 'alert',
    styles: {
      background: '#feefe2',
      color: '#78471c',
      borderColor: '#ba6e2c'
    }
  },
  error: {
    image: <Image icon="alert" color="red" />,
    label: 'Error',
    icon: 'alert',
    styles: {
      background: '#fbe5e5',
      color: '#9b3134'
    }
  },
  beta: {
    image: <Image icon="lightning" color="blue" />,
    label: 'Beta',
    icon: 'lightning',
    tooltipText: 'This feature is in public beta and is subject to changes.',
    styles: {
      background: '#f1f3fd',
      color: '#0c248d',
      borderColor: '#0428c8'
    }
  },
  new: {
    image: <Image icon="plus" color="green" />,
    label: 'New!',
    icon: 'plus',
    tooltipText: 'This feature was released recently.',
    styles: {
      background: '#e8f5ee',
      color: '#15603d',
      borderColor: '#378645'
    }
  },
  download: {
    image: <Image icon="arrow-down" color="purple" />,
    label: 'Download',
    icon: 'arrow-down',
    styles: {
      background: '#f2effa',
      color: '#452f92'
    }
  },
  fundamentals: {
    label: 'Fundamentals',
    icon: 'bookmark',
    tooltipText:
      'The concepts described here are fundamental to using this product.',
    styles: {
      background: '#fff0f7',
      color: '#cf1c61',
      borderColor: '#fd8ac0'
    }
  }
};
/* duplicate themes */
themes.legacy = Object.assign({}, themes.warning);
themes.legacy.label = 'Legacy';
themes.legacy.tooltipText = 'This feature is no longer in active development.';

export default themes;
