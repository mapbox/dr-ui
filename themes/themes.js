import React from 'react';
import PropTypes from 'prop-types';
import Icon from '@mapbox/mr-ui/icon';
class Image extends React.PureComponent {
  render() {
    const {
      color,
      icon
    } = this.props;
    return /*#__PURE__*/React.createElement("div", {
      className: `bg-${color}-lighter round-full color-${color} flex flex--center-main flex--center-cross`,
      style: {
        width: 50,
        height: 50
      }
    }, /*#__PURE__*/React.createElement(Icon, {
      name: icon,
      size: 40
    }));
  }
}
const themes = {
  default: {
    image: /*#__PURE__*/React.createElement(Image, {
      icon: "book",
      color: "gray"
    }),
    label: 'NOTE',
    icon: 'book',
    background: 'bg-gray-faint'
  },
  warning: {
    image: /*#__PURE__*/React.createElement(Image, {
      icon: "alert",
      color: "orange"
    }),
    label: 'WARNING',
    icon: 'alert',
    background: 'bg-orange-faint',
    color: 'color-orange-dark',
    border: 'border--orange-dark'
  },
  error: {
    image: /*#__PURE__*/React.createElement(Image, {
      icon: "alert",
      color: "red"
    }),
    label: 'ERROR',
    icon: 'alert',
    background: 'bg-red-faint'
  },
  beta: {
    image: /*#__PURE__*/React.createElement(Image, {
      icon: "lightning",
      color: "blue"
    }),
    label: 'BETA',
    icon: 'lightning',
    tooltipText: 'This feature is in public beta and is subject to changes.',
    styles: {
      backgroundColor: '#CBE4FF'
    }
  },
  new: {
    image: /*#__PURE__*/React.createElement(Image, {
      icon: "plus",
      color: "green"
    }),
    label: 'NEW',
    icon: 'plus',
    tooltipText: 'This feature was released recently.',
    styles: {
      backgroundColor: '#C9F2DD'
    }
  },
  download: {
    image: /*#__PURE__*/React.createElement(Image, {
      icon: "arrow-down",
      color: "purple"
    }),
    label: 'DOWNLOAD',
    icon: 'arrow-down',
    background: 'bg-purple-faint'
  },
  fundamentals: {
    label: 'FUNDAMENTALS',
    icon: 'bookmark',
    tooltipText: 'The concepts described here are fundamental to using this product.',
    background: 'bg-pink-faint'
  },
  pricing: {
    image: /*#__PURE__*/React.createElement(Image, {
      icon: "creditcard",
      color: "green"
    }),
    label: 'PRICING',
    tooltipText: 'This contains information about product pricing.',
    background: 'bg-green-faint'
  }
};
/* duplicate themes */
themes.legacy = Object.assign({}, themes.warning);
themes.legacy.label = 'LEGACY';
themes.legacy.tooltipText = 'This feature is no longer in active development.';
export default themes;