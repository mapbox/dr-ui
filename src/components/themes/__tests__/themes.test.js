import renderer from 'react-test-renderer';
import ColorContrastChecker from 'color-contrast-checker';
import { testCases } from './themes-test-cases';
import themes from '../themes';

const ccc = new ColorContrastChecker();

describe('themes', () => {
  Object.keys(themes).forEach((theme) => {
    describe(`theme: ${theme}`, () => {
      test('has label', () => {
        expect(themes[theme].label).toBeDefined();
        expect(
          themes[theme].label[0].toUpperCase() === themes[theme].label[0]
        ).toBeTruthy();
      });

      if (themes[theme].tooltipText) {
        test('tooltipText ends with period', () => {
          expect(themes[theme].tooltipText.endsWith('.')).toBeTruthy();
        });
      }

      test('has styles', () => {
        expect(themes[theme].styles).toBeDefined();
        expect(typeof themes[theme].styles).toBe('object');
        expect(themes[theme].styles.background).toBeDefined();
        expect(themes[theme].styles.color).toBeDefined();
      });

      test(`styles color and background contrast is WCAG AA compliant`, () => {
        expect(
          ccc.isLevelAA(
            themes[theme].styles.color,
            themes[theme].styles.background
          )
        ).toBeTruthy();
      });
    });
  });

  describe(testCases.default.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.default;
      wrapper = renderer.create(testCase.element);
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.error.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.error;
      wrapper = renderer.create(testCase.element);
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.download.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.download;
      wrapper = renderer.create(testCase.element);
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.beta.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.beta;
      wrapper = renderer.create(testCase.element);
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.legacy.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.legacy;
      wrapper = renderer.create(testCase.element);
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.fundamentals.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.fundamentals;
      wrapper = renderer.create(testCase.element);
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.custom.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.custom;
      wrapper = renderer.create(testCase.element);
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });
});
