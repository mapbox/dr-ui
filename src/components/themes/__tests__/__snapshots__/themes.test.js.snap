// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`themes Custom tag renders as expected 1`] = `
<div>
  <div
    className="inline-block"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
  >
    <div
      aria-describedby="tooltip-7"
      className="txt-bold px6 round inline-block cursor-default txt-s border"
      style={
        Object {
          "background": "#FEDADA",
          "borderColor": "#FD8383",
          "color": "#DA2E30",
        }
      }
    >
      Limited access
    </div>
    <div
      className="hide-visually"
      id="tooltip-7"
      role="tooltip"
    >
      <div
        className="txt-s wmax120"
      >
        Contact us for access to this feature.
      </div>
    </div>
  </div>
  <div
    className="inline-block"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
  >
    <div
      aria-describedby="tooltip-8"
      className="txt-bold px6 round inline-block cursor-default txt-xs"
      style={
        Object {
          "background": "#FEDADA",
          "borderColor": "#FD8383",
          "color": "#DA2E30",
        }
      }
    >
      Limited access
    </div>
    <div
      className="hide-visually"
      id="tooltip-8"
      role="tooltip"
    >
      <div
        className="txt-s wmax120"
      >
        Contact us for access to this feature.
      </div>
    </div>
  </div>
</div>
`;

exports[`themes Default renders as expected 1`] = `
<div
  className="dr-ui--note py18 px18 round flex-parent flex-parent--row mb18"
  style={
    Object {
      "background": "#f4f7fb",
      "color": "#425870",
    }
  }
>
  <div
    className="flex-child mr18 none block-mm pt3"
  >
    <div
      className="bg-gray-light round-full color-gray flex-parent flex-parent--center-main flex-parent--center-cross"
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      }
    >
      <svg
        className="events-none icon"
        focusable="false"
        role="presentation"
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      >
        <use
          xlinkHref="#icon-book"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
    </div>
  </div>
  <div
    className="flex-child prose"
  >
    <div
      className="txt-bold txt-m mb6"
    >
      Note
    </div>
    Default note
  </div>
</div>
`;

exports[`themes beta renders as expected 1`] = `
<div>
  <div
    className="dr-ui--note py18 px18 round flex-parent flex-parent--row mb18"
    style={
      Object {
        "background": "#f1f3fd",
        "color": "#0c248d",
      }
    }
  >
    <div
      className="flex-child mr18 none block-mm pt3"
    >
      <div
        className="bg-blue-light round-full color-blue flex-parent flex-parent--center-main flex-parent--center-cross"
        style={
          Object {
            "height": 50,
            "width": 50,
          }
        }
      >
        <svg
          className="events-none icon"
          focusable="false"
          role="presentation"
          style={
            Object {
              "height": 40,
              "width": 40,
            }
          }
        >
          <use
            xlinkHref="#icon-marker"
            xmlnsXlink="http://www.w3.org/1999/xlink"
          />
        </svg>
      </div>
    </div>
    <div
      className="flex-child prose"
    >
      <div
        className="txt-bold txt-m mb6"
      >
        Beta
      </div>
      Beta note.
    </div>
  </div>
   
  <div
    className="inline-block"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
  >
    <div
      aria-describedby="tooltip-1"
      className="txt-bold px6 round inline-block cursor-default txt-s border"
      style={
        Object {
          "background": "#f1f3fd",
          "borderColor": "#0428c8",
          "color": "#0c248d",
        }
      }
    >
      Beta
    </div>
    <div
      className="hide-visually"
      id="tooltip-1"
      role="tooltip"
    >
      <div
        className="txt-s wmax120"
      >
        This feature is in public beta and is subject to changes.
      </div>
    </div>
  </div>
   
  <div
    className="inline-block"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
  >
    <div
      aria-describedby="tooltip-2"
      className="txt-bold px6 round inline-block cursor-default txt-xs"
      style={
        Object {
          "background": "#f1f3fd",
          "borderColor": "#0428c8",
          "color": "#0c248d",
        }
      }
    >
      Beta
    </div>
    <div
      className="hide-visually"
      id="tooltip-2"
      role="tooltip"
    >
      <div
        className="txt-s wmax120"
      >
        This feature is in public beta and is subject to changes.
      </div>
    </div>
  </div>
</div>
`;

exports[`themes download renders as expected 1`] = `
<div
  className="dr-ui--note py18 px18 round flex-parent flex-parent--row mb18"
  style={
    Object {
      "background": "#f2effa",
      "color": "#452f92",
    }
  }
>
  <div
    className="flex-child mr18 none block-mm pt3"
  >
    <div
      className="bg-purple-light round-full color-purple flex-parent flex-parent--center-main flex-parent--center-cross"
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      }
    >
      <svg
        className="events-none icon"
        focusable="false"
        role="presentation"
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      >
        <use
          xlinkHref="#icon-arrow-down"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
    </div>
  </div>
  <div
    className="flex-child prose"
  >
    <div
      className="txt-bold txt-m mb6"
    >
      Download
    </div>
    download note
  </div>
</div>
`;

exports[`themes error renders as expected 1`] = `
<div
  className="dr-ui--note py18 px18 round flex-parent flex-parent--row mb18"
  style={
    Object {
      "background": "#fbe5e5",
      "color": "#9b3134",
    }
  }
>
  <div
    className="flex-child mr18 none block-mm pt3"
  >
    <div
      className="bg-red-light round-full color-red flex-parent flex-parent--center-main flex-parent--center-cross"
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      }
    >
      <svg
        className="events-none icon"
        focusable="false"
        role="presentation"
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      >
        <use
          xlinkHref="#icon-alert"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
    </div>
  </div>
  <div
    className="flex-child prose"
  >
    <div
      className="txt-bold txt-m mb6"
    >
      Error
    </div>
    error note
  </div>
</div>
`;

exports[`themes fundamentals renders as expected 1`] = `
<div>
  <div
    className="inline-block"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
  >
    <div
      aria-describedby="tooltip-5"
      className="txt-bold px6 round inline-block cursor-default txt-s border"
      style={
        Object {
          "background": "#fff0f7",
          "borderColor": "#fd8ac0",
          "color": "#cf1c61",
        }
      }
    >
      Fundamentals
    </div>
    <div
      className="hide-visually"
      id="tooltip-5"
      role="tooltip"
    >
      <div
        className="txt-s wmax120"
      >
        The concepts described here are fundamental to using this product.
      </div>
    </div>
  </div>
  <div
    className="inline-block"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
  >
    <div
      aria-describedby="tooltip-6"
      className="txt-bold px6 round inline-block cursor-default txt-xs"
      style={
        Object {
          "background": "#fff0f7",
          "borderColor": "#fd8ac0",
          "color": "#cf1c61",
        }
      }
    >
      Fundamentals
    </div>
    <div
      className="hide-visually"
      id="tooltip-6"
      role="tooltip"
    >
      <div
        className="txt-s wmax120"
      >
        The concepts described here are fundamental to using this product.
      </div>
    </div>
  </div>
</div>
`;

exports[`themes legacy or warning renders as expected 1`] = `
<div>
  <div
    className="dr-ui--note py18 px18 round flex-parent flex-parent--row mb18"
    style={
      Object {
        "background": "#feefe2",
        "color": "#78471c",
      }
    }
  >
    <div
      className="flex-child mr18 none block-mm pt3"
    >
      <div
        className="bg-orange-light round-full color-orange flex-parent flex-parent--center-main flex-parent--center-cross"
        style={
          Object {
            "height": 50,
            "width": 50,
          }
        }
      >
        <svg
          className="events-none icon"
          focusable="false"
          role="presentation"
          style={
            Object {
              "height": 40,
              "width": 40,
            }
          }
        >
          <use
            xlinkHref="#icon-alert"
            xmlnsXlink="http://www.w3.org/1999/xlink"
          />
        </svg>
      </div>
    </div>
    <div
      className="flex-child prose"
    >
      <div
        className="txt-bold txt-m mb6"
      >
        Legacy
      </div>
      legacy note
    </div>
  </div>
  <div
    className="inline-block"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
  >
    <div
      aria-describedby="tooltip-3"
      className="txt-bold px6 round inline-block cursor-default txt-s border"
      style={
        Object {
          "background": "#feefe2",
          "borderColor": "#ba6e2c",
          "color": "#78471c",
        }
      }
    >
      Legacy
    </div>
    <div
      className="hide-visually"
      id="tooltip-3"
      role="tooltip"
    >
      <div
        className="txt-s wmax120"
      >
        This feature is no longer in active development.
      </div>
    </div>
  </div>
  <div
    className="inline-block"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
  >
    <div
      aria-describedby="tooltip-4"
      className="txt-bold px6 round inline-block cursor-default txt-xs"
      style={
        Object {
          "background": "#feefe2",
          "borderColor": "#ba6e2c",
          "color": "#78471c",
        }
      }
    >
      Legacy
    </div>
    <div
      className="hide-visually"
      id="tooltip-4"
      role="tooltip"
    >
      <div
        className="txt-s wmax120"
      >
        This feature is no longer in active development.
      </div>
    </div>
  </div>
</div>
`;
